File: src/main/kotlin/web/TaskController.kt
package web

import dao.TaskDao
import javax.validation.Valid
import model.Task
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Controller
import org.springframework.ui.Model
import org.springframework.validation.BindingResult
import org.springframework.web.bind.annotation.ModelAttribute
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestMethod
import org.springframework.web.bind.annotation.RequestParam

Controller
RequestMapping(value = array("/task"))
public class TaskController [Autowired] (
        val dao:TaskDao
    ) {

    RequestMapping
    public fun list(model: Model): String {
        model.addAttribute("list", dao.findAll())
        return "list"
    }

    RequestMapping(params = array("edit"))
    public fun form(): String = "form"

    RequestMapping(method = array(RequestMethod.POST))
    public fun save(Valid task:Task, result: BindingResult): String {
        return when (result.hasErrors()) {
            true -> "form"
            else -> {
                dao.save(task)
                "redirect:task?edit=${task.id}"
            }
        }
    }

    ModelAttribute
    public fun model(RequestParam(required = false) edit: Int?): Task? {
        return when (edit) {
            null -> Task(null, "", "")
            else -> dao.findById(edit)
        }
    }
}

File: scala/src/main/scala/web/TaskController.scala
package web

import dao.TaskDao
import javax.validation.Valid
import model.Task
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Controller
import org.springframework.ui.Model
import org.springframework.validation.BindingResult
import org.springframework.web.bind.annotation.ModelAttribute
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestMethod
import org.springframework.web.bind.annotation.RequestParam

@Controller
@RequestMapping(Array("/task"))
class TaskController @Autowired() (val dao: TaskDao) {

  @RequestMapping
  def list(model: Model): String = {
    model.addAttribute("list", dao.findAll())
    "list"
  }

  @RequestMapping(params = Array("edit"))
  def form() = "form"

  @RequestMapping(method = Array(RequestMethod.POST))
  def save(@Valid task: Task, result: BindingResult): String = {
    result.hasErrors match {
      case true => "form"
      case false => {
        dao.save(task)
        "redirect:task?edit=" + task.id
      }
    }
  }

  @ModelAttribute
  def model(@RequestParam(required = false) edit: Option[Int]): Option[Task] = {
    edit match {
      case Some(x) => dao.findById(x)
      case None => Option(new Task)
    }
  }
}